Xamarin.Forms.Platform.Android/CollectionView/CarouselViewRenderer.cs

```diff
@@ -183,6 +184,28 @@ protected override void UpdateAdapter()
 		void UpdateInitialPosition()
 		{
 			_initialPosition = Carousel.Position;
+
+			if (Carousel.CurrentItem != null)
+			{
+				int position = 0;
+
+				var items = Carousel.ItemsSource as IList;
```

Comment:
ItemsSource shouldn't be assumed to be an IList. This has caused other problems in the past with ListViews and CollectionViews. The type is defined as an IEnumerable.